File: go/base_feature.go
package voxgigsolardemosdk

type BaseFeature struct {
	Version string
	Name    string
	Active  bool
}

func NewBaseFeature() *BaseFeature {
	return &BaseFeature{
		Version: "0.0.1",
		Name:    "base",
		Active:  true,
	}
}

func (f *BaseFeature) GetVersion() string { return f.Version }
func (f *BaseFeature) GetName() string    { return f.Name }
func (f *BaseFeature) GetActive() bool    { return f.Active }

func (f *BaseFeature) Init(ctx *Context, options map[string]any)  {}
func (f *BaseFeature) PostConstruct(ctx *Context)                 {}
func (f *BaseFeature) PostConstructEntity(ctx *Context)           {}
func (f *BaseFeature) SetData(ctx *Context)                       {}
func (f *BaseFeature) GetData(ctx *Context)                       {}
func (f *BaseFeature) GetMatch(ctx *Context)                      {}
func (f *BaseFeature) SetMatch(ctx *Context)                      {}
func (f *BaseFeature) PreTarget(ctx *Context)                     {}
func (f *BaseFeature) PreSelection(ctx *Context)                  {}
func (f *BaseFeature) PreSpec(ctx *Context)                       {}
func (f *BaseFeature) PreRequest(ctx *Context)                    {}
func (f *BaseFeature) PreResponse(ctx *Context)                   {}
func (f *BaseFeature) PreResult(ctx *Context)                     {}
func (f *BaseFeature) PreDone(ctx *Context)                       {}
func (f *BaseFeature) PreUnexpected(ctx *Context)                 {}
func (f *BaseFeature) PostOperation(ctx *Context)                 {}

File: php/feature/BaseFeature.php
<?php
declare(strict_types=1);

// Solardemo SDK base feature

class SolardemoBaseFeature
{
    public string $version;
    public string $name;
    public bool $active;

    // Positions this feature when added via the client `extend` option:
    // "__before__" / "__after__" / "__replace__" name an already-added
    // feature (mirrors the ts feature `_options`). Declared so setting it
    // on an extension instance avoids the dynamic-property deprecation.
    public ?array $_options = null;

    public function __construct()
    {
        $this->version = '0.0.1';
        $this->name = 'base';
        $this->active = true;
    }

    public function get_version(): string { return $this->version; }
    public function get_name(): string { return $this->name; }
    public function get_active(): bool { return $this->active; }

    public function init(SolardemoContext $ctx, array $options): void {}
    public function PostConstruct(SolardemoContext $ctx): void {}
    public function PostConstructEntity(SolardemoContext $ctx): void {}
    public function SetData(SolardemoContext $ctx): void {}
    public function GetData(SolardemoContext $ctx): void {}
    public function GetMatch(SolardemoContext $ctx): void {}
    public function SetMatch(SolardemoContext $ctx): void {}
    public function PrePoint(SolardemoContext $ctx): void {}
    public function PreSpec(SolardemoContext $ctx): void {}
    public function PreRequest(SolardemoContext $ctx): void {}
    public function PreResponse(SolardemoContext $ctx): void {}
    public function PreResult(SolardemoContext $ctx): void {}
    public function PreDone(SolardemoContext $ctx): void {}
    public function PreUnexpected(SolardemoContext $ctx): void {}
}
